content/source/docs/glossary.html.md

```diff
@@ -0,0 +1,940 @@
+
+In Terraform's [configuration][] language: a piece of syntax that represents a value, either literally or by referencing and combining other values. Expressions appear as values for [arguments][], or within other expressions.
+
+- [Terraform docs: Expressions](/docs/configuration/expressions.html)
+
+## Fork
+
+[fork]: glossary.html#fork
+[forks]: glossary.html#fork
+[forked]: glossary.html#fork
+
+Also "forked repository" or "forked repo".
+
+A [VCS][] [repository][] that was created by copying content and history from another repository.
+
+Different VCS providers handle forks differently, but a fork is usually owned by a different person or organization than the original repo, and a fork usually does not inherit all of the original repo's access permissions.
+
+Terraform Enterprise makes extensive use of VCS repos, and assumes that forks of a trusted repo are not necessarily trusted. As such, TFE avoids evaluating any code from external forks, which prevents TFE from running [speculative plans][] for [pull requests][] from forks.
+
+## Git
+
+[git]: glossary.html#git
+
+A distributed [version control system][vcs] for tracking changes in source code during software development. It is designed for coordinating work among programmers, but it can be used to track changes in any set of files.
+
+- [Wikipedia: Git](https://en.wikipedia.org/wiki/Git)
+
+## HCL
+
+[hcl]: glossary.html#hcl
+
+HashiCorp Configuration Language. The structured configuration syntax that serves as the basis for Terraform's [configuration][] language, as well as the configuration layer for several other HashiCorp products.
+
+HCL establishes the syntax Terraform uses for things like [arguments][], [blocks][], literal values, and [expressions][]. But what most people think of as the Terraform language extends beyond just the syntax; the built-in functions, Terraform-specific block types (like `resource` and `variable`), and Terraform-specific named values available in expressions are all implementation details of Terraform itself.
+
+- [GitHub: HCL](https://github.com/hashicorp/hcl)
+- [Terraform docs: Configuration Language](/docs/configuration/index.html)
+
+## ID
+
+[id]: glossary.html#id
+[ids]: glossary.html#id
+
+An identifier; an opaque string permanently associated with a specific object, which doesn't contain any information about the object but which can be used to retrieve information about it or perform actions on it via an [API][].
+
+In Terraform, many [resource][] types have an ID [attribute][] that helps link Terraform's [state][] to the real infrastructure resource being managed.
+
+In Terraform Enterprise, most internal application objects (like [workspaces][], users, [policies][], etc.) can be identified by both a name (possibly including the parents, for disambiguation) and by an opaque, permanent ID. Most API endpoints use IDs instead of names, since names sometimes change. IDs for TFE's application objects are sometimes called "external IDs."
```

Comment:
That, yeah. Probably unnecessary.

```suggestion
In Terraform Enterprise, most internal application objects (like [workspaces][], users, [policies][], etc.) can be identified by both a name and by an opaque, permanent ID. Most API endpoints use IDs instead of names, since names sometimes change. IDs for TFE's application objects are sometimes called "external IDs."
```